### ISSUE TITLE
Undefined symbol "OPENSSL_no_config"

### ISSUE THREAD

**Issue by janev94** [pyca/cryptography#5209]
Hi,

 Trying to use a module that uses the crypto package I get `ImportError: /usr/local/lib/python3.7/site-packages/cryptography/hazmat/bindings/_openssl.abi3.so: Undefined symbol "OPENSSL_no_config"`. Seems like in newer versions of openssl OPENSSL_no_config was deprecated in favour of OPENSSL_init.

Related info:

**OS version**: FreeBSD 11.3-RELEASE #0 r349754: Fri Jul  5 04:45:24 UTC 2019

**Cryptography version**: cryptography==2.9
**cffi version**: cffi==1.14.0
**OpenSSL version**: OpenSSL 1.0.2s-freebsd  28 May 2019
**PIP version**: pip 20.0.2 
**setuptools version**: setuptools 44.0.0 

Used pip to install cryptography. To reproduce the issue use the mentioned OS. Install the official openssl & pip versions and attempt to run http3server from: https://github.com/aiortc/aioquic.

**Comment by reaperhulk** [comment]
`OPENSSL_no_config` is defined in `conf.h` in OpenSSL 1.0.2s, so something is going on here with your include and linker paths. Based on the error it seems likely you've got a copy of OpenSSL 1.1.0+ somewhere on your system that it's linking against, but you're using the headers for 1.0.2.

**Comment by janev94** [comment]
You are correct. FreeBSD's default binary for openssl is 1.0.2s but this is not registered as a package. So installing a newer one via pkg (currently 1.1.1f,1 is available) will not update the original.... Long story short, had to upgrade my openssl the hard way - by re-compiling it, but now that I have both the binary and the headers for 1.1.1f,1 everything seems to work.
